internal/controller/git/git.go

```diff
@@ -86,25 +121,24 @@ type Repo interface {
 	// contains any differences from what's already at the head of the current
 	// branch.
 	HasDiffs() (bool, error)
-	// GetDiffPaths returns a string slice indicating the paths, relative to the
-	// root of the repository, of any new or modified files.
-	GetDiffPaths() ([]string, error)
-	// GetDiffPathsForCommitID returns a string slice indicating the paths, relative to the
-	// root of the repository, of any files that are new or modified since the given commit ID.
-	GetDiffPathsSinceCommitID(commitId string) ([]string, error)
+	// GetDiffPathsForCommitID returns a string slice indicating the paths,
+	// relative to the root of the repository, of any files that are new or
+	// modified in the commit with the given ID.
+	GetDiffPathsForCommitID(commitID string) ([]string, error)
 	// IsAncestor returns true if parent branch is an ancestor of child
 	IsAncestor(parent string, child string) (bool, error)
 	// LastCommitID returns the ID (sha) of the most recent commit to the current
 	// branch.
 	LastCommitID() (string, error)
-	// ListTags returns a slice of tags in the repository.
-	ListTags() ([]string, error)
+	// ListTagsWithMetadata returns a slice of tags in the repository with metadata
+	// such as commit ID, creator date, and subject.
+	ListTagsWithMetadata() ([]TagMetadata, error)
+	// ListCommitsWithMetadata returns a slice of commits in the current branch
+	// with metadata such as commit ID, commit date, and subject.
+	ListCommitsWithMetadata(limit, skip uint) ([]CommitMetadata, error)
```

Comment:
Nit: Since I don't think we need to differentiate from any functions that list Tags or Commits _without_ metadata, do we want to rename these to just `ListTags()` and `ListCommits()`?

Whatever you decide is fine with me.

Reply:
Agree on this one, will ammend.